### ISSUE TITLE
Getting `java.lang.NullPointerException` for a query.

### ISSUE THREAD

**Issue by monicaluodialpad** [apache/pinot#10824]
## Issue
Getting
```
ProcessingException(errorCode:150, message:SQLParsingError:
java.lang.NullPointerException: _Fields.STRING_VALUE
	at java.base/java.util.Objects.requireNonNull(Objects.java:246)
	at org.apache.pinot.common.request.Literal.setStringValue(Literal.java:601)
	at org.apache.pinot.common.utils.request.RequestUtils.getLiteralExpression(RequestUtils.java:124)
	at org.apache.pinot.sql.parsers.CalciteSqlParser.toExpression(CalciteSqlParser.java:676))
```
when I wanted to try some non-aggregated functions
```
SELECT CASE WHEN "some_field" IS NOT NULL THEN "some_field" ELSE NULL END AS "some_alias"
FROM "some_table"
```
The type of "some_field" can be String, Int, Float, List etc.

--------------
## Use Case of this Query:
Wanted to explicitly return Null if it is Null value, instead of returning default value (0, or MIN_INTEGER or 'null' ) to outside.

**Comment by Jackie-Jiang** [comment]
@shenyu0127 Can you please take a look?

**Comment by shenyu0127** [comment]
If you run a Pinot that is built on HEAD, you will not get this error.

Details:
- You were running a Pinot that is a few months old. (The line 601 of `Literal.java` is `Objects.requireNonNull` means it is before https://github.com/apache/pinot/pull/10217).
- I tested your query using a Pinot built on HEAD and it worked. You may need to add `OPTION(enableNullHandling=true)` in the query.

**Comment by monicaluodialpad** [comment]
@shenyu0127 Thank you for your reply
* May I ask the full query example with `OPTION(enableNullHandling=true)`? We have turned it on in table configuration. Is it good enought?
* Is the latest version of pinot 0.12.1 ? Unfortunately I still get the error.

**Comment by shenyu0127** [comment]
- If you put a `;` after your `SET` statement, your query will be parsed correctly.
- Your query will not work in v0.12.1.
   - The `IS NOT NULL` feature is added in https://github.com/apache/pinot/pull/10594, which is not included in v0.12.1. 
   - If you build a Pinot from source on HEAD (https://docs.pinot.apache.org/basics/getting-started/running-pinot-locally), your query will work. Or you can wait for the next Pinot release to include this feature.

**Comment by monicaluodialpad** [comment]
Closing this as it is working on pinot 1.0.0
